pkg/api/mlflow/dao/repositories/metric.go

```diff
@@ -85,8 +85,19 @@ func (r MetricRepository) CreateBatch(
 		lastIters[lastMetric.Key] = lastMetric.LastIter
 	}
 
+	uniqueContexts := make([]*models.Context, 0, len(metrics))
+	contextMap := make(map[string]*models.Context)
 	latestMetrics := make(map[string]models.LatestMetric)
 	for n, metric := range metrics {
+		if metric.Context != nil {
+			hash := getJsonHash(metric.Context.Json)
+			if contextMap[hash] == nil {
```

Comment:
something like `if foundCtx := contextMap[hash]; foundCtx == nil {` might be slightly more efficient since you can then  use `foundCtx` at line 98